src/Features/CustomerEffortScoreTracks.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * WooCommerce Customer effort score tracks
+ *
+ * @package WooCommerce\Admin\Features
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Triggers customer effort score on several different actions.
+ */
+class CustomerEffortScoreTracks {
+	/**
+	 * Class instance.
+	 *
+	 * @var CustomerEffortScoreTracks instance
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Get class instance.
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor. Sets up filters to hook into WooCommerce.
+	 */
+	public function __construct() {
+		add_filter( 'add_meta_boxes_product', array( $this, 'add_meta_boxes_product' ) );
+	}
+
+	/**
+	 * Hook into the Publish button for products, to trigger the customer
+	 * effort score widget after save.
+	 *
+	 * @param WP_Post $post The post, not used.
+	 */
+	public function add_meta_boxes_product( $post ) {
+		wc_enqueue_js(
```

Comment:
Interesting - would it be possible to use the product lifecycle hooks here at all? This made me curious to how we do something similar after a product is added as part of the setup checklist, and the approach there is somewhat similar but no localStorage used: https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks.php#L233

Reply:
(Kinda) done in df92a4634e8ee6dc0a8b26fb3f711c6b6fb251b3